lib/rucio/daemons/storage/consistency/actions.py

```diff
@@ -0,0 +1,771 @@
+
+    def save(self):
+        try:
+            with open(self.Path, "r") as f:
+                data = f.read()
+        except:
+            #traceback.print_exc()
+            data = ""
+        #print("data:", data)
+        data = json.loads(data or "{}")
+        data.update(self.Data)
+        open(self.Path, "w").write(json.dumps(data, indent=4))
+
+
+def write_stats(my_stats, stats_file, stats_key=None):
+    if stats_file:
+        stats = {}
+        if os.path.isfile(stats_file):
+            with open(stats_file, "r") as f:
+                stats = json.loads(f.read())
+        if stats_key:
+            stats[stats_key] = my_stats
+        else:
+            stats.update(my_stats)
+        open(stats_file, "w").write(json.dumps(stats))
+
+
+
+def cmp2dark(new_list="T2_US_Purdue_2021_06_18_02_28_D.list",\
+             old_list="T2_US_Purdue_2021_06_17_02_28_D.list",\
+             comm_list="out_D.list", stats_file="test_stats.json"):
+
+    t0 = time.time()
+    stats_key = "cmp2dark"
+    my_stats = stats = None
+    op = "and"
+
+    with open(new_list, "r") as a_list, open(old_list, "r") as b_list,\
+         open(comm_list, "w") as out_list:
+
+        if stats_file is not None:
+            stats = Stats(stats_file)
+            my_stats = {
+                "elapsed": None,
+                "start_time": t0,
+                "end_time": None,
+                "new_list": new_list,
+                "old_list": old_list,
+                "out_list": out_list.name,
+                "status": "started"
+            }
+            stats[stats_key] = my_stats
+
+        a_set = set(line.strip() for line in a_list)
+        b_set = set(line.strip() for line in b_list)
+
+
+# The intersection of the two sets is what can be deleted
+        out_set = a_set & b_set
+#        print("\n\n\n\n out_set: \n\n\n", "\n".join( sorted(list(out_set)) ))
+
+        out_list.writelines("\n".join(sorted(list(out_set))))
+
+    t1 = time.time()
+
+    if stats_file:
+        my_stats.update({
+            "elapsed": t1-t0,
+            "end_time": t1,
+            "status": "done"
+        })
+        stats[stats_key] = my_stats
+###############################
+### This was Igor's Stats class
+###############################
+
+
+####################
+### from Deckard
+####################
+def parse_filename(fn):
+    # filename looks like this:
+    #
+    #   <rse>_%Y_%m_%d_%H_%M_<type>.<extension>
+    #
+    fn, ext = fn.rsplit(".", 1)
+    parts = fn.split("_")
+    typ = parts[-1]
+    timestamp_parts = parts[-6:-1]
+    timestamp = "_".join(timestamp_parts)
+    rse = "_".join(parts[:-6])
+    return rse, timestamp, typ, ext
+
+def list_cc_scanned_rses(Path):
+    files = glob.glob(f"{Path}/*_stats.json")
+    rses = set()
+    for path in files:
+        fn = path.rsplit("/", 1)[-1]
+        rse, timestamp, typ, ext = parse_filename(fn)
+        rses.add(rse)
+    return sorted(list(rses))
+
+def list_runs_by_age(Path, rse, reffile):
```

Comment:
Maybe too much for this round, but all these functions taking Path as a first argument (why is Path upper case?) are a good candidate for a class

Reply:
The capital P is a left-over from the deckard code. Will correct it.
Grouping them into a class is a good idea for future development/improvement.